Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1784,5 +1760,71 @@ public InvalidBlockException ValidateNextBlock(Block<T> block)
 
             return null;
         }
+
+        /// <summary>
+        /// Checks if given <paramref name="block"/> is a valid genesis <see cref="Block{T}"/>.
+        /// </summary>
+        /// <param name="block">The target <see cref="Block{T}"/> to validate.</param>
+        /// <returns><see langword="null"/> if given <paramref name="block"/> is valid,
+        /// otherwise an <see cref="InvalidBlockException"/>.</returns>
+        /// <exception cref="ArgumentException">If <paramref name="block"/> has
+        /// <see cref="Block{T}.Index"/> value anything other than 0.</exception>
+        private static InvalidBlockException ValidateGenesisBlock(Block<T> block)
+        {
+            if (block.Index != 0)
+            {
+                throw new ArgumentException(
+                    $"Given {nameof(block)} must have index 0 but has index {block.Index}",
+                    nameof(block));
+            }
+
+            int actualProtocolVersion = block.ProtocolVersion;
+            const int currentProtocolVersion = Block<T>.CurrentProtocolVersion;
+            if (block.ProtocolVersion > Block<T>.CurrentProtocolVersion)
+            {
+                return new InvalidBlockProtocolVersionException(
+                    $"The protocol version ({actualProtocolVersion}) of the block " +
+                    $"#{block.Index} {block.Hash} is not supported by this node." +
+                    $"The highest supported protocol version is {currentProtocolVersion}.",
+                    actualProtocolVersion);
+            }
+
+            if (block.PreviousHash is { } previousHash)
+            {
+                return new InvalidBlockPreviousHashException(
+                    $"A genesis block should not have previous hash, " +
+                    $"but its value is {block.PreviousHash}.");
+            }
+
+            return null;
+        }
```

Comment:
We probably need `LastCommit` validation somewhere in here. 🙃

Reply:
You mean `GenesisBlock` should have `null` `LastCommit`?

Reply:
Yeah, this bypasses the rest of `ValidateNextBlock()` so it won't be checked.

Reply:
Great, added in https://github.com/planetarium/libplanet/pull/2871/commits/65e1e13cad23d7fe9c1b98b03bf2d6a538dd2f4b.